apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2KeyspacesResourceImpl.java

```diff
@@ -111,11 +121,32 @@ public Uni<RestResponse<Sgv2NameResponse>> createKeyspace(final String payloadSt
               .build();
     }
 
+    // Simple execution but resulting failure from bad DC needs to be mapped to something
+    // more useful (as per [stargate#2231])
     return executeQueryAsync(query)
+        .onFailure()
+        .transform(
+            failure ->
+                failureMatchesBadDC(failure, dcMap)
+                    ? new WebApplicationException(
+                        String.format(
+                            "Bad request: one or more of datacenters passed (%s) invalid",
+                            dcMap.keySet()),
+                        Response.Status.BAD_REQUEST)
+                    : failure)
         // No real contents; can ignore ResultSet it seems and only worry about exceptions
         .map(any -> restResponseCreatedWithName(keyspaceName));
   }
 
+  private boolean failureMatchesBadDC(Throwable t, Map<String, Integer> dcMap) {
```

Comment:
@tlasica Yes, I am not happy with having to rely on failure message, but at least match is specific so there should not be false positives.

But the real problem in checking for existence is that code here has no direct access to this information: all it can do is talk to Stargate Coordinator via Bridge API that only allows queries. I didn't think DC information was available through queries (perhaps it is); but more generally it is simpler operationally to attempt an operation, only validating what can be done locally and react to failure messages.
This is one of challenges with Stargate V2, now that APIs are separate from coordinators so there is much less access to specifics of persistence backend.

I agree that it'd be great to have simple access to existing DCs to pre-validate this particular case.

Reply:
I have proposed some simplified approach to just treat `ConfigurationExceptions` in the special way instead of trying to find a specific "unknown DC" one.

I am sorry, I did not realize that there is no access to the information about the DCs.

Reply:
@tlasica No I appreciate your help & suggestions. Handling also changed a lot from Stargate V1 (where APIs had direct access in-process to backend) to V2 (where gRPC call is needed).

Another problem then is that API does not get `ConfigurationException` directly but something Bridge gRPC translates. Worse, Bridge handles all queries about the same (no separate end point for creating keyspaces just general "executeQuery()") so adding better translation is tricky.

I'll keep the ideas in mind and I hope at some point we can improve on this. Default handling will just pass exception message REST API gets to caller: perhaps that could suggest that DC information should be double-checked.

Finally: it's all bit moot wrt Astra too: turns out Stargate keyspace creation is blocked currently so users will actually get very different fail message from OSS Stargate :)